### ISSUE TITLE
PopupMenuItem Unknown Right padding on `tapDown`

### ISSUE THREAD

**Issue by LIMMIHEE** [flutter/flutter#138335]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Use a PopupMenuItem and padding parameter set EdgeInsets.zero

### Expected results

PopupMenuItem show without Padding

<img width="312" alt="image" src="https://github.com/flutter/flutter/assets/48482259/5b637da5-5723-41bf-b7b5-be2345c0bdc5">

### Actual results

PopupMenuItem with Unknown Right Padding

https://github.com/flutter/flutter/assets/48482259/d15d5f66-da92-44ee-8035-552ad2edbabb

### Code sample

<font class="papago-parent"><font class="papago-source" style="display:none;"><details><summary>Code sample</summary>
</font></font>
<font class="papago-parent"><font class="papago-source" style="display:none;">
</font></font>

<font class="papago-parent"><font class="papago-source" style="display:none;">
```dart
import 'package:flutter/material.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
          backgroundColor: const Color(0xFFFFFFFF),
          body: SizedBox(
            width: 120,
            child: PopupMenuButton(
                elevation: 0,
                color: Colors.transparent,
                padding: EdgeInsets.zero,
                onSelected: (value) {},
                itemBuilder: (context) => [
                      PopupMenuItem(
                          padding: EdgeInsets.zero,
                          value: 'write_work',
                          child: Container(
                            color: const Color(0xFF6C6051),
                            child: const Text('add\nwork',
                                style: TextStyle(color: Color(0xFF000000))),
                          )),
                    ],
                icon:
                    const Icon(Icons.edit, size: 28, color: Color(0xFF000000))),
          )),
    );
  }
}
```
</font></font>

<font class="papago-parent"><font class="papago-source" style="display:none;">
</font>
</font><font class="papago-parent"><font class="papago-source" style="display:none;"></details>
</font></details></font>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<font class="papago-parent"><font class="papago-source" style="display:none;"><details><summary>Doctor output</summary>
</font></font>

<font class="papago-parent"><font class="papago-source" style="display:none;">
</font>
</font>

<font class="papago-parent"><font class="papago-source" style="display:none;">

```
[✓] Flutter (Channel stable, 3.13.4, on macOS 13.4 22F66 darwin-arm64, locale ko-KR)
    • Flutter version 3.13.4 on channel stable at /Users/
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (9 weeks ago), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/
    • Platform android-33, build-tools 33.0.2
    • ANDROID_SDK_ROOT = /Users/
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.84.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 8CE0EB23-A5A1-4DD1-BBCC-849C7D40A673 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4
      (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

• No issues found! 

```

</font></font>

<font class="papago-parent"><font class="papago-source" style="display:none;">
</font>
</font><font class="papago-parent"><font class="papago-source" style="display:none;"></details>
</font></details></font>

**Comment by dam-ease** [comment]
Thanks for filing this @LIMMIHEE.
Can you help with more information on the expected behaviour here?
I can reproduce the actual behaviour as described here on the latest versions of flutter

**Comment by LIMMIHEE** [comment]
@dam-ease 

After testing, the problem occurred even when I used Continer instead of Text, and if I deleted the alignment parameter from the Widget item under the PopupMenuItemState<T, W extension PopupMenuItem<T>> class code build function below, the child width increased by the right padding.

Even if I deleted the other parameters, the right padding remained the same as before.


You can set the PopupMenuButton constraints parameter to prevent padding, but that doesn't solve the underlying problem.

**Comment by dam-ease** [comment]
Thanks for your response!
Please can you help with a screen record of the actual behaviour vs. Expected behaviour?

**Comment by LIMMIHEE** [comment]
I'm checking with dart Pad, but it's not working as shown in the Without Padding video. Could you please provide the code?

The screen recording you requested is as follows.
<br/><br/><br/>

## actual behaviour
<br/>

**With EdgeInsets.zero**

https://github.com/flutter/flutter/assets/48482259/eb5c375c-a32a-41ba-bc20-1b4d2992d42b

<br/>

**Without Padding**
Even if you Without padding of PopupMenuButton, it is the same screen.

https://github.com/flutter/flutter/assets/48482259/755c61cc-2765-41dd-b1cf-c34308ec1f48


<br/>
<br/>
<br/>
<br/>

## Expected behaviour
I applied the constraints property of PopupMenuButton to create the screen I expected and recorded it.


https://github.com/flutter/flutter/assets/48482259/fac57d57-2556-49c3-841f-f7c7c38af671

**Comment by dam-ease** [comment]
Thanks for your detailed response. I see the issue you are referring to is when the PopupMenuItem is `tappedDown`, I can reproduce this on the latest `master` and `stable` channels as well even when `SizedBox` widget is removed.

<details><summary>stable, master flutter doctor -v</summary>
<p>

```
[!] Flutter (Channel stable, 3.13.9, on macOS 14.0 23A344 darwin-arm64, locale
    en-NG)
    • Flutter version 3.13.9 on channel stable at
      /Users/damilolaalimi/sdks/flutter
    ! Warning: `dart` on your path resolves to
      /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/damilolaalimi/sdks/flutter.
      Consider adding /Users/damilolaalimi/sdks/flutter/bin to the front of your
      path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (3 weeks ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio
      Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (5 available)
    • sdk gphone64 arm64 (mobile)        • emulator-5554
      • android-arm64  • Android 14 (API 34) (emulator)
    • Damilola’s iPhone (mobile)         • 00008110-001964480AE1801E
      • ios            • iOS 17.0.2 21A351
    • iPad Air (5th generation) (mobile) • B3716F7D-587E-4473-9F87-F1BA94E31B29
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)                    • macos
      • darwin-arm64   • macOS 14.0 23A344 darwin-arm64
    • Chrome (web)                       • chrome
      • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 
```
[!] Flutter (Channel master, 3.17.0-9.0.pre.22, on macOS 14.0 23A344 darwin-arm64, locale en-NG)
    • Flutter version 3.17.0-9.0.pre.22 on channel master at /Users/damilolaalimi/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.1.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/damilolaalimi/fvm/versions/master. Consider adding /Users/damilolaalimi/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a930fd8e4a (50 minutes ago), 2023-11-15 00:03:08 -0500
    • Engine revision a7f2267dd1
    • Dart version 3.3.0 (build 3.3.0-133.0.dev)
    • DevTools version 2.30.0-dev.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/damilolaalimi/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[!] Android Studio (version unknown)
    • Android Studio at /Users/damilolaalimi/Downloads/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] VS Code (version 1.83.1)
    • VS Code at /Users/damilolaalimi/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (5 available)
    • sdk gphone64 arm64 (mobile)        • emulator-5554                        • android-arm64  • Android 14 (API 34) (emulator)
    • Damilola’s iPhone (mobile)         • 00008110-001964480AE1801E            • ios            • iOS 17.0.2 21A351
    • iPad Air (5th generation) (mobile) • B3716F7D-587E-4473-9F87-F1BA94E31B29 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)                    • macos                                • darwin-arm64   • macOS 14.0 23A344 darwin-arm64
    • Chrome (web)                       • chrome                               • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
``` 

</p>
</details>

**Comment by TahaTesser** [comment]
Hi @LIMMIHEE 

The "Right padding" you see is the minimum popup menu item width.  If the child width is smaller than the minimum item width then the item will have a minimum.

https://github.com/flutter/flutter/blob/e9de4484209472079d3db1cc0dfd1fcf54570b65/packages/flutter/lib/src/material/popup_menu.dart#L37-L40

> I applied the constraints property of PopupMenuButton to create the screen I expected and recorded it.

This is exactly it is designed for. 

From https://api.flutter.dev/flutter/material/PopupMenuButton/constraints.html docs:

> The default constraints ensure that the menu width matches maximum width recommended by the Material Design guidelines. Specifying this parameter enables creation of menu wider than the default maximum width.

You can use `constraints` property  for your purpose. When you use it you're overriding the minimum width,


https://github.com/flutter/flutter/blob/e9de4484209472079d3db1cc0dfd1fcf54570b65/packages/flutter/lib/src/material/popup_menu.dart#L623-L627


Closing
